CMakeLists.txt

```diff
@@ -3,7 +3,22 @@
 # SPDX-License-Identifier: Apache-2.0
 
 cmake_minimum_required(VERSION 3.22)
-project(aws-greengrass-sdk-lite C)
+
+# Read version from file
+file(READ version GGSDKL_VERSION)
+string(STRIP "${GGSDKL_VERSION}" GGSDKL_VERSION)
+
+project(
+  aws-greengrass-sdk-lite
+  VERSION ${GGSDKL_VERSION}
+  LANGUAGES C)
+
+# Parse version components
+string(REPLACE "." ";" VERSION_LIST ${GGSDKL_VERSION})
+list(GET VERSION_LIST 0 GGSDKL_VERSION_MAJOR)
+
+# Add version to compile definitions
+add_compile_definitions("GGSDKL_VERSION=\"${GGSDKL_VERSION}\"")
```

Comment:
```suggestion
```

Cmake parses version for us. Also we should not set preprocessor definitions we are not using.